api/src/test/java/org/openmrs/module/pacsintegration/outgoing/OrderToPacsComponentTest.java

```diff
@@ -12,44 +12,50 @@
  * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
  */
 
-package org.openmrs.module.pacsintegration.component;
-
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
-import org.junit.Before;
-import org.junit.Test;
-import org.mockito.ArgumentMatcher;
-import org.mockito.Mockito;
+package org.openmrs.module.pacsintegration.outgoing;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
 import org.openmrs.Encounter;
+import org.openmrs.Order;
 import org.openmrs.TestOrder;
 import org.openmrs.api.EncounterService;
 import org.openmrs.api.LocationService;
+import org.openmrs.api.OrderService;
 import org.openmrs.api.ProviderService;
 import org.openmrs.api.context.Context;
-import org.openmrs.module.pacsintegration.NonTransactionalBaseModuleContextSensitiveTest;
+import org.openmrs.event.Event;
+import org.openmrs.module.pacsintegration.PacsIntegrationConstants;
 import org.openmrs.module.pacsintegration.api.PacsIntegrationService;
-import org.openmrs.module.pacsintegration.listener.OrderEventListener;
+import org.openmrs.module.pacsintegration.runner.ContextTaskRunner;
+import org.openmrs.module.pacsintegration.test.TransactionalTestService;
 import org.openmrs.module.radiologyapp.RadiologyOrder;
-import org.openmrs.module.radiologyapp.RadiologyService;
+import org.openmrs.test.jupiter.BaseModuleContextSensitiveTest;
 import org.springframework.beans.factory.annotation.Autowired;
 
+import javax.jms.MapMessage;
 import java.text.SimpleDateFormat;
 import java.util.Date;
 
+import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.startsWith;
-import static org.junit.Assert.assertThat;
-import static org.mockito.Matchers.any;
-import static org.mockito.Matchers.argThat;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.timeout;
 
-public class OrderToPacsComponentTest extends NonTransactionalBaseModuleContextSensitiveTest {
+public class OrderToPacsComponentTest extends BaseModuleContextSensitiveTest {
 
     @Autowired
     private EncounterService encounterService;
 
+    @Autowired
+    private OrderService orderService;
+
+    @Autowired
+    private TransactionalTestService transactionalTestService;
```

Comment:
To get this test working, I needed to force OpenMRS to commit a transaction with the encounter details, in order to get the event module's hibernate interceptor to fire and an event to be send to the listeners.  To do that, I created a new service in the test project that has a REQUIRES_NEW transactional annotation, and then used that to save the encounter instead of going straight to the encounter service (which participates in the existing transaction and is never committed).